src/Core/Services/MetadataProviders/MsSqlMetadataProvider.cs

```diff
@@ -79,5 +79,24 @@ public override async Task PopulateTriggerMetadataForTable(string entityName, st
                 }
             }
         }
+
+        /// <inheritdoc/>
+        public override bool TryResolveDbType(string sqlDbTypeName, out DbType dbType)
+        {
+            if (Enum.TryParse(sqlDbTypeName, ignoreCase: true, out SqlDbType sqlDbType))
+            {
+                // For MsSql, all the date time types i.e. date, smalldatetime, datetime, datetime2 map to System.DateTime system type.
+                // Hence we cannot directly determine the DbType from the system type.
+                // However, to make sure that the database correctly interprets these datatypes, it is necessary to correctly
+                // populate the DbTypes.
+                return TypeHelper.TryGetDbTypeFromSqlDbDateTimeType(sqlDbType, out dbType);
+            }
+
+            // This code should never be hit because every sqlDbTypeName must have a corresponding sqlDbType.
+            // However, when a new data type is introduced in MsSql which maps to .NET type of DateTime, this code block
+            // will be hit. Returning false instead of throwing an exception in that case prevents the engine from crashing.
+            dbType = 0;
```

Comment:
What does dbType 0 correspond to?

Reply:
<img width="148" alt="image" src="https://github.com/Azure/data-api-builder/assets/34566234/220f5516-df2b-40ad-9f86-958197a22597">

We don't care here though since we are returning false in which case we don't use the `out` value